File: TurkSoft.WebUI/wwwroot/ApiJS/entities/KullaniciMaliMusavir.js
// entities/KullaniciMaliMusavir.js
// ------------------------------------------------------------------
// Kullanıcı ↔ MaliMüşavir pivotu için REST sarmalayıcı (standart CRUD).
// Route: /api/v1/KullaniciMaliMusavir
// ------------------------------------------------------------------
import BaseApiService from '../Base/BaseAPIService.js';

const svc = new BaseApiService('KullaniciMaliMusavir');

// Standart CRUD
export const list = (p = null) => svc.list(p);
export const get = (id) => svc.get(id);
export const create = (d) => svc.create(d);   // {KullaniciId,MaliMusavirId,IsPrimary,...}
export const update = (id, d) => svc.update(id, d);
export const remove = (id) => svc.remove(id);

export default { list, get, create, update, remove };

File: TurkSoft.Entities/EntityDB/KullaniciMaliMusavir.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TurkSoft.Entities.EntityDB
{
    /// <summary>Kullanıcı ↔ MaliMüşavir pivotu</summary>
    public class KullaniciMaliMusavir : BaseEntity
    {
        public Guid KullaniciId { get; set; }
        public Kullanici? Kullanici { get; set; } = null!;

        public Guid MaliMusavirId { get; set; }
        public MaliMusavir? MaliMusavir { get; set; } = null!;

        public bool IsPrimary { get; set; } = false;
        public string? AtananRol { get; set; }           // "MMAdmin","Stajyer","Personel" vb.
        public DateTime? BaslangicTarihi { get; set; }
        public DateTime? BitisTarihi { get; set; }
    }
}
